shuup/admin/modules/media/views.py

```diff
@@ -64,18 +69,69 @@ def _filer_folder_to_json_dict(folder, children=None):
     }
 
 
+def _ensure_media_folder(shop, folder):
+    media_folder, created = MediaFolder.objects.get_or_create(folder=folder)
+    if not media_folder.shops.filter(id=shop.id).exists():
+        media_folder.shops.add(shop)
+
+
+def _is_folder_shared(folder):
+    if not settings.SHUUP_ENABLE_MULTIPLE_SHOPS:
+        return False
+
+    media_folder = MediaFolder.objects.filter(folder=folder).first()
+    if not media_folder:
+        return True
+    return bool(media_folder.shops.count() > 1)
+
+
+def _get_folder_query_filter(shop):
+    return Q(Q(media_folder__isnull=True) | Q(media_folder__shops=shop))
+
+
+def _get_folder_query(shop, folder=None):
+    queryset = Folder.objects.filter(_get_folder_query_filter(shop))
+    if folder:
+        queryset = queryset.filter(id=folder.id)
+    return queryset
+
+
+def _ensure_media_file(shop, file):
+    media_file, created = MediaFile.objects.get_or_create(file=file)
+    if not media_file.shops.filter(id=shop.id).exists():
+        media_file.shops.add(shop)
+
+
+def _is_file_shared(file):
+    if not settings.SHUUP_ENABLE_MULTIPLE_SHOPS:
+        return False
+
+    media_file = MediaFile.objects.filter(file=file).first()
+    if not media_file:
+        return True
+    return bool(media_file.shops.count() > 1)
```

Comment:
same above. the commit description says that if the shops is null, it is shared so you should also consider when `count() == 0`